doc/development/guide/architecture.rst

```diff
@@ -0,0 +1,228 @@
+
+.. list-table::
+   :widths: 35 45 60 
+   :header-rows: 1
+
+   * - **Device**
+     - **Shortname**
+     - **Description**
+   * - :class:`~.DefaultGaussian`
+     - ``"default.gaussian"``
+     - Default gaussian device
+   * - `Strawberry Fields Fock device <https://pennylane-sf.readthedocs.io/en/latest/devices/fock.html>`__
+     - ``"strawberryfields.fock"``
+     - Fock device giving access to the Strawberry Fields Fock state simulator backend
+   * - `Strawberry Fields Gaussian device <https://pennylane-sf.readthedocs.io/en/latest/devices/gaussian.html>`__
+     - ``"strawberryfields.gaussian"``
+     - Gaussian device giving access to the Strawberry Fields Fock state simulator backend
+
+
+:html:`</div></div>`
+
+QNodes
+******
+
+A  quantum  node or ``QNode`` (represented by a subclass to
+:class:`~.BaseQNode`) is an encapsulation of a function :math:`f(x;θ):R^m→R^n`
+that is executed using quantum information processing on a quantum
+device.
+
+Each ``QNode`` represents the quantum circuit by building a
+:class:`~.CircuitGraph` instance, but the way differentiation is done is custom
+to the differentiation method offered by the ``QNode``.
+
+For further details on qnodes, the mutability of qnodes we refer to
+:class:`~.BaseQNode` and for a list of qnodes to :ref:`qml_qnodes`.
+
+Interfaces
+**********
+
+The integration between classical and quantum computations is encompassed by
+interfaces.
+
+We refer to the :ref:`intro_interfaces` page for a more in-depth introduction
+and a list of available interfaces.
+
+Optimizers
+**********
+
+Optimizers are objects which can be used to automatically update the parameters
+of a quantum or hybrid machine learning model.
+
+We refer to the :ref:`intro_ref_opt` page for a more in-depth introduction
+and a list of available optimizers.
+
+Key design details
+##################
+
+The following are key design details related to how PennyLane works internally.
+
+Queuing of operators
+********************
+
+In PennyLane, the construction of quantum gates is separated from the specific
+quantum node (:class:`~.BaseQNode`) that they belong to. However, including
+logic for this when creating an instance of :class:`~.Operator` does not align
+with the current architecture. Therefore, there is a need to use a high-level
+object that holds information about the relationship between quantum gates and
+a quantum node.
+
+The :class:`~.QueuingContext` class realizes this by providing access to the current
+QNode.  Furthermore, it provides the flexibility to have multiple objects
+record the creation of quantum gates.
+
+The ``QueuingContext`` class both acts as the abstract base class for all
+classes that expose a queue for Operations (so-called contexts), as well as the
+interface to said queues. The active contexts contain maximally one QNode and
+an arbitrary number of other contexts like the :class:`~.OperationRecorder`.
```

Comment:
mostly preferred placing pointer links such that we only have to maintain a text at one place. This one is an exception as the text is also part of the docstring of the QueuingContext class. (was unsure how this could be referenced easily there :thinking:)